### ISSUE TITLE
Prometheus annotations are misplaced

### ISSUE THREAD

**Issue by stefanprodan** [traefik/traefik-helm-chart#460]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik-helm-chart/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What version of the Traefik's Helm Chart are you using?

10.1.1

### What version of Traefik are you using?

2.4.9

### What did you do?

Installing v10.1.1 with the default values that should enable Prometheus scraping on port 9100, but since the annotations are misplaced, scraping never happens.

### What did you see instead?

The Prometheus annotations are misplaced, they should be placed on pods not on the deployment.

### What is your environment & configuration?

Kubernetes Kind 

### Additional Information

```shell
Workaround here: https://github.com/fluxcd/flagger/pull/960
```

**Comment by z0rc** [comment]
Also reported at https://github.com/traefik/traefik-helm-chart/pull/452#issuecomment-883169697

**Comment by js-timbirkett** [comment]
It's also sort of annoying that Prometheus has been configured in the default `values.yaml` so even if a user isn't making use of Prometheus it configures Traefik to expose (and collect) the metrics.

**Comment by js-timbirkett** [comment]
It also looks like the metrics port is created in such a way that it is always there whether you want it or not...
https://github.com/traefik/traefik-helm-chart/blob/master/traefik/values.yaml#L298-L311

**Comment by faust64** [comment]
Indeed, having Prometheus enabled by default in values sounds wrong.

Setting up Prometheus integration used to be doable, just configuring proper annotations (.Values.deployment.podAnnotations, .Values.service.annotations), labels (.Values.deployment.podLabels, .Values.service.labels) and extra arguments (.Values.globalArguments), documentation was clear enough what needed to be done, ...

Not long ago, #415 was reminding us how those ports can not be disabled, once added into the Charts defaults.
Nowadays, in addition to the web / websecure / traefik ports, which we could argue should always be installed: we have a new prometheus port, that can't be removed. Setting `.Values.metrics.prometheus=false`, you would still end up generating a container with a Prometheus entrypoint and containerPort declared.

And while on that topic, another matter would be whether relying on Prometheus pod discovery is relevant.
IMHO, it only makes sense probing containers that may expose metrics while unready. Otherwise: services/endpoints roles are safer, as they would automatically discard unhealthy pods. Dealing with larger Prometheus deployments, it's going to have an impact on the time required for Prometheus to iterate over all its probes.
If anything, Prometheus integration samples should have been left commented out, giving both pod & service label samples, maybe explaining why use one or the other.

**Comment by SantoDE** [comment]
I think the initial issue is fixed now, right?

For @faust64 comments:
```
Setting up Prometheus integration used to be doable, just configuring proper annotations (.Values.deployment.podAnnotations, .Values.service.annotations), labels (.Values.deployment.podLabels, .Values.service.labels) and extra arguments (.Values.globalArguments), documentation was clear enough what needed to be done, ...
```
While I agree, we saw that people didn't figure it. So I guess, that's an easy way to enable it now :)

```
Not long ago, #415 was reminding us how ports from the default values file can not be disabled.
Nowadays, in addition to the web / websecure / traefik ports, which we could argue should always be installed: we have a new prometheus port, that can't be removed. Setting .Values.metrics.prometheus=false, you would still end up generating a container with a Prometheus entrypoint and containerPort declared.
```
I think, that's a general discussion we will need to take at one point soon :)

**Comment by jakubhajek** [comment]
The issue has been fixed in the PR #478
